### DISCUSSION TITLE
Feature Request: add support for GitHub-Issue-Based Comments on both Posts and Pages

### DISCUSSION THREAD

**Discussion by ElectricRCAircraftGuy** [mmistakes/minimal-mistakes#2714] [Ideas]
## Summary

I'd like to see support in your Jekyll themes for Github-Issue-based comments. That means that you open up a GitHub issue as the means of posting comments on any given page or post. See here for a demo: https://jekyllcodex.org/blog/gdpr-compliant-comment/, and here for more setup info: https://jekyllcodex.org/without-plugin/comments/.

## Motivation

I'd like to see this supported because it has some tremendous advantages over other comment systems:

1. No cost. Anyone can create a GitHub account and comment for free, and there's not cost to opening up an issue to be used for comments on your already-no-cost blog or website using GitHub pages.
1. Easy editing of comments and markdown support: the GitHub issue editor has markdown support while also supporting basic formatting buttons for those not familiar with markdown. For those who ARE familiar with markdown, however, it allows beautiful code discussion by placing code blocks right into the comment!
1. Easy moderation: as the owner of a repo, you can easily go in and modify, delete, or moderate comments as needed.
1. Traceable. Since you can easily quote comments, add links, etc, the comment page (a GitHub Issue) is nicely traceable to see the full comment history and gain a better understanding of what's going on.

## Drawbacks

1. You have to leave the blog page to edit or add your comment.
1. There may be a limit on number of queries to the issue as people refresh the comments on the site?
1. Requires a github account. If your site is about programming (many are) and caters to other developers, this is great, not bad! But, if your user is the average person, they won't know what GitHub is and may prefer to use a Facebook login or something instead of creating a GitHub account to comment.
1. It looks like the visitor has to manually click a refresh link to show new comments, even after new ones are written.

**Comment by mmistakes** [comment]
Have a look at the comments documentation for the theme. It supports **[Utterances](https://github.com/apps/utterances)** which does what you want by using your repo's Issues as a place to store post comments.

https://mmistakes.github.io/minimal-mistakes/docs/configuration/#utterances-comments
